resources/append_if_no_line.rb

```diff
@@ -4,36 +4,18 @@
 resource_name :append_if_no_line
 
 action :edit do
-  string = escape_string new_resource.line
+  string = Regexp.escape(new_resource.line)
   regex = /^#{string}$/
 
-  if ::File.exist?(new_resource.path)
-    begin
-      f = ::File.open(new_resource.path, 'r+')
+  current = ::File.readlines(new_resource.path)
+  # we match the regexp after doing this append for files without terminating CRs.  should
+  # we instead match against the unchanged content?  we're basically saying "don't worry
+  # about terminating CRs or not, we gotcha covered" which feels like the 99% use case.  but
+  # is there a 1% use case here which considers this a bug?
+  current[-1] = current[-1].chomp + "\n"
 
-      found = false
-      f.each_line { |line| found = true if line =~ regex }
-
-      unless found
-        converge_by "Updating file #{new_resource.path}" do
-          f.puts new_resource.line
-        end
-      end
-    ensure
-      f.close
-    end
-  else
-    begin
-      f = ::File.open(new_resource.path, 'w')
-      converge_by "Updating file #{new_resource.path}" do
-        f.puts new_resource.line
-      end
-    ensure
-      f.close
-    end
+  file new_resource.path do
+    content((current + [new_resource.line + "\n"]).join)
```

Comment:
Whole file in memory vs line by line, some chance putting the file in memory will break for someone.

Reply:
rewriting very large files that will fill up the ram seems not ideal anyway. And probably outside the main use cases of line imho.

Reply:
I was thinking more in terms of version 2.0.0 vs a minor version bump and fixing bug reports if we got any.  I've thought of rewriting this the way your did for years but never did because I figured someone would object to slurping the file.